### PR TITLE
Bug/#213/uhfqc detector normalization

### PR DESCRIPTION
**Pull request by AdriaanRol**
@NielsBultink I have updated the detectors for the UFHQC to support the different acquisition modes. 

Can I ask you to look at the changes in the detector functions file in detail? Also please don't merge if OK as this still needs to be tested (both on your setup (tek_transmon)  and mine (CC_transmon). 

Here is a summary of the changes. 
- new result_logging parameter as part of the UFHQC detectors that supersedes the crosstalk suppression parameter. See summary below. 
- Added docstrings and comments that should clarify some things 
- updated the relevant files (qubit objects multi qubit module) to work with this update 
- added the multiplexed RO analysis file based on the code by Niels (should help addressing #178 ) 
- DDM if statement was defined twice in tek transmon object, so I removed one. 
- I removed the rotate and normalize option from the UHFQC detector as this is replaced with the optimal weight normalized detector. 

## Result logging parameter change
The detectors now have a parameter "result_logging_mode" that has three options. 
Below is an excerpt from the docstring: 
```
        result_logging_mode (str) :  options are
            - raw        -> returns raw data in V
            - normalized -> returns normalized data using the crosstalk
                            suppression matrix and offset defined in the
                            UHFQC instrument. Requires optimal weights.
            - digitized  -> returns fraction of shots based on the threshold
                            defined in the UFHQC. Requires optimal weights.
```

If one wants to calibrate the SSRO threshold and inversion matrix, the result_logging_mode should be set to normalized with the correction matrix set to the Identity matrix. 

I have

### CODE REVIEW

File: examples/qubit_tuneup/CPhase_tuneup.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
               DataT.RO_acq_weight_function_I()],
     nr_averages=AncT.RO_acq_averages(),
     integration_length=AncT.RO_acq_integration_length(),
-    cross_talk_suppression=True)
+    result_logging_mode='normalized')
```
Comment (NielsBultink) [inline examples/qubit_tuneup/CPhase_tuneup.py]: as discussed with @AdriaanRol: change 'normalized' to 'lintrans' as it is a general linear transformation, not just normalization.

**Comment by AdriaanRol** [comment]
@NielsBultink , Renamed "normalized" mode to "lin_trans" as discussed. Includes update to docstring.

File: pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py [hunk 2]
```diff
@@ -103,7 +103,8 @@ def measure_spectroscopy(self):
     def measure_transients(self):
         raise NotImplementedError()
 
-    def calibrate_optimal_weights(self):
+    def calibrate_optimal_weights(self, MC=None, verify=True,
```
Comment (NielsBultink) [inline pycqed/instrument_drivers/meta_instrument/qubit_objects/qubit_object.py]: I would prefer to have a more general 'set_weights( )' function. Based on three qubit object parameters(all booleans) a different calibration/setting is performed:
- SSB: determins whether we use single sideband or double sideband measurement
- optimized weights: determines whether weight functions are measured or standard weight functions are set 
- one weight functions, only loads weight functions into the I channel

**Comment by AdriaanRol** [comment]
@NielsBultink small status update. 
I have tested the changes to the detector on the CC_transmon. This is now working. 
Before we can merge this, several things need to be done. 
- [x] Test on CBox/CC driven transmon 
- [ ] Test on tektronix driven transmon (relates to merging your branch)
- [ ] cleanup new correlations detector to fit in the same paradigm 
- [ ] resolve #212 as using only one weight doesn't play well with analysis. 

Let me know if I missed something (and if this is OK with what you had in mind based on our discussion earlier).

**Comment by AdriaanRol** [comment]
@NielsBultink I would like to merge this as it's been in the backlog for quite a while. I think we've discussed most of the things in here. 
Further updates are on the Starmon branch but I would like to have that in a separate review.

**Pull request merged**